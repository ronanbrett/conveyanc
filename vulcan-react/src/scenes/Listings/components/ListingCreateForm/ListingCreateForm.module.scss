@import "src/styles/vars";

.ListingCreateForm {
  padding: var(--spacing-xl) var(--spacing-xl);
  overflow-y: auto;
  :global .title {
    @include type-header-l;
    margin-bottom: var(--spacing-m);
  }
}
