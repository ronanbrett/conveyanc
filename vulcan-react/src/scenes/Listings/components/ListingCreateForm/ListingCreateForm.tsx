import {
  Input,
  MultiTierDropdown,
  MultiTierDropdownItem,
  MultiTierDropdownOption,
} from "@components";
import { PropertyInfo } from "@core/api/graphql";
import { ErrorMessage, Field, FieldProps, Form, Formik } from "formik";
import { groupBy } from "lodash-es";
import React, { FC, ReactElement } from "react";
import styles from "./ListingCreateForm.module.scss";

import { object, string } from "yup";

const ListingCreateFormSchema = object().shape({
  propertyType: string().required("You must select a type"),
  name: string().required("You must add a name"),
});

interface ListingCreateFormProps {
  fieldData?: PropertyInfo;
  children?: any;
}

const ListingCreateForm: FC<ListingCreateFormProps> = ({
  fieldData,
  ...props
}) => {
  const propertyTypeOptions = groupBy(fieldData?.propertyType, "group");

  const getPropertyTypeOptions = (items: any[]) => {
    let content: ReactElement[] = [];
    items.forEach((item, index) =>
      content.push(
        <MultiTierDropdownOption key={index} value={item.value}>
          {item.label}
        </MultiTierDropdownOption>
      )
    );

    return content;
  };

  const getPropertyTypeItems = () => {
    let content: ReactElement[] = [];
    for (let key in propertyTypeOptions) {
      const items: any[] = propertyTypeOptions[key];
      content.push(
        <MultiTierDropdownItem key={key} value={key} label={key}>
          {getPropertyTypeOptions(items)}
        </MultiTierDropdownItem>
      );
    }
    return content;
  };

  const onSubmit = (formValues: any) => {
    console.log(formValues);
  };

  return (
    <div className={styles.ListingCreateForm}>
      <Formik
        validationSchema={ListingCreateFormSchema}
        onSubmit={onSubmit}
        initialValues={{ propertyType: "DUPLEX", name: "" }}
      >
        <Form className="form">
          <div className="field__container">
            <label className="field__label" htmlFor="propertyType">
              Name
            </label>
            <MultiTierDropdown
              placeholder="Select Property Type"
              name="propertyType"
            >
              {getPropertyTypeItems()}
            </MultiTierDropdown>
            <ErrorMessage name="propertyType">
              {(msg) => <div>{msg}</div>}
            </ErrorMessage>
          </div>

          <div className="field__container">
            <label className="field__label" htmlFor="propertyType">
              Name
            </label>

            <Input id="name" name="name" placeholder="John" />

            <ErrorMessage name="name">{(msg) => <div>{msg}</div>}</ErrorMessage>
          </div>

          <button type="submit">Submit</button>
          <button type="reset">Reset</button>
        </Form>
      </Formik>
    </div>
  );
};

export default ListingCreateForm;
