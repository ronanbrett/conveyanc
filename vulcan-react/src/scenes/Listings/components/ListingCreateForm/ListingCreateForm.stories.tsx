import React from 'react';

import ListingCreateForm from './ListingCreateForm';

import { action } from '@storybook/addon-actions';

export default {
  title: 'Components/ListingCreateForm',
  decorators: [],
  component: ListingCreateForm,
};


export const Default = () => (
  <ListingCreateForm></ListingCreateForm>
);
