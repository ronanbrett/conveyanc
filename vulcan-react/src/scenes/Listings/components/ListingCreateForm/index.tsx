import ListingCreateForm from './ListingCreateForm';

export default ListingCreateForm;