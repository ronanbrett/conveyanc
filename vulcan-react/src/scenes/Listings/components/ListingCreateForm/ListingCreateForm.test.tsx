import React from "react";
import { render } from "@testing-library/react";
import ListingCreateForm from "./ListingCreateForm";
import { create } from "react-test-renderer";

const generatePortal = () => {
  let modalRoot = document.createElement("div");
  modalRoot.setAttribute("id", "topNavPortal");
  document.querySelector("body")!.appendChild(modalRoot);
};

describe("ListingCreateForm", () => {
  beforeAll(() => {
    generatePortal();
    window.getSelection = () => {
      return {} as any;
    };
  });
  it("renders without error", () => {
    const { container } = render(<ListingCreateForm />);
    expect(container).toMatchSnapshot();
  });
});
