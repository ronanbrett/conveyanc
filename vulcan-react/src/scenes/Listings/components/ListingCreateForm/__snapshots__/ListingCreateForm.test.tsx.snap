// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListingCreateForm renders without error 1`] = `
<div>
  <div
    class="ListingCreateForm"
  >
    <h1
      class="title"
    >
      Property Details
    </h1>
    <form
      action="#"
      class="form"
    >
      <div
        class="field__container"
      >
        <label
          class="field__label"
          for="propertyType"
        >
          Type of Property
        </label>
        <div
          class="multi-dd field--valid"
        >
          <header
            class="multi-dd__trigger"
          >
            <div
              class="multi-dd__trigger-content"
            >
              Select Property Type
            </div>
            <button
              class="icon-button icon-button--default mulit-dd__trigger-btn"
              type="button"
            >
              <div
                class="icon__container"
              >
                <i
                  class="material-icons"
                >
                  keyboard_arrow_down
                </i>
                <div
                  class="icon__background"
                />
              </div>
            </button>
          </header>
          <div
            class="multi-dd__content hidden"
          />
        </div>
      </div>
      <div
        class="field__container"
      >
        <label
          class="field__label"
          for="eircode"
        >
          Eircode
        </label>
        <div>
          <div
            class="CharacterInput"
          >
            <div
              class="TextInput__input__container TextInput--single"
              style="transform: none;"
            >
              <input
                autocomplete="off"
                class="TextInput__input TextInput--text"
                maxlength="1"
                pattern="[A-Za-z0-9]{1}"
                type="text"
                value=""
              />
            </div>
            <div
              class="TextInput__input__container TextInput--single"
              style="transform: none;"
            >
              <input
                autocomplete="off"
                class="TextInput__input TextInput--text"
                maxlength="1"
                pattern="[A-Za-z0-9]{1}"
                type="text"
                value=""
              />
            </div>
            <div
              class="TextInput__input__container TextInput--single"
              style="transform: none;"
            >
              <input
                autocomplete="off"
                class="TextInput__input TextInput--text"
                maxlength="1"
                pattern="[A-Za-z0-9]{1}"
                type="text"
                value=""
              />
            </div>
            <div
              class="TextInput__input__container TextInput--single"
              style="transform: none;"
            >
              <input
                autocomplete="off"
                class="TextInput__input TextInput--text"
                maxlength="1"
                pattern="[A-Za-z0-9]{1}"
                type="text"
                value=""
              />
            </div>
            <div
              class="TextInput__input__container TextInput--single"
              style="transform: none;"
            >
              <input
                autocomplete="off"
                class="TextInput__input TextInput--text"
                maxlength="1"
                pattern="[A-Za-z0-9]{1}"
                type="text"
                value=""
              />
            </div>
            <div
              class="TextInput__input__container TextInput--single"
              style="transform: none;"
            >
              <input
                autocomplete="off"
                class="TextInput__input TextInput--text"
                maxlength="1"
                pattern="[A-Za-z0-9]{1}"
                type="text"
                value=""
              />
            </div>
            <div
              class="TextInput__input__container TextInput--single"
              style="transform: none;"
            >
              <input
                autocomplete="off"
                class="TextInput__input TextInput--text"
                maxlength="1"
                pattern="[A-Za-z0-9]{1}"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="field__container"
      >
        <label
          class="field__label"
          for="propertyType"
        >
          Address
        </label>
        <div
          class="TextInput__input__container TextInput--medium"
        >
          <input
            autocomplete="off"
            class="TextInput__input TextInput--text"
            id="name"
            name="address"
            placeholder="Enter the first line of your address"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="field__container"
      >
        <div
          class="field__label"
        >
          Description
        </div>
        <div
          class="RichTextEditor"
        >
          <div
            contenteditable="true"
            data-gramm="false"
            data-slate-editor="true"
            data-slate-node="value"
            role="textbox"
            spellcheck="true"
            style="outline: none; white-space: pre-wrap; word-wrap: break-word;"
          >
            <p
              data-slate-node="element"
            >
              <span
                data-slate-node="text"
              >
                <span
                  data-slate-leaf="true"
                >
                  <span
                    contenteditable="false"
                    style="pointer-events: none; display: inline-block; width: 0px; max-width: 100%; white-space: nowrap; opacity: 0.333; user-select: none; font-style: normal; font-weight: normal; text-decoration: none;"
                  >
                    Tell us about the Property
                  </span>
                  <span
                    data-slate-length="0"
                    data-slate-zero-width="n"
                  >
                    ï»¿
                    <br />
                  </span>
                </span>
              </span>
            </p>
          </div>
        </div>
      </div>
      <div
        class="field__container"
      >
        <label
          class="field__label"
          for="propertyType"
        >
          Images
        </label>
        <div
          class="ImageUploader"
        >
          <input
            accept="image/*"
            multiple=""
            style="display: none;"
            type="file"
          />
          <div
            class="ImageUploader__wrapper"
          >
            <button
              class="ImageUploader__drop"
              type="button"
            >
              Click or Drop here
            </button>
            <div
              class="ImageUploader__images"
            />
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;
